packages/safe-tools-client/app/components/truncated-blockchain-address/index.gts

```diff
@@ -0,0 +1,78 @@
+import Component from '@glimmer/component';
+import BoxelIconButton from '@cardstack/boxel/components/boxel/icon-button';
+import { on } from '@ember/modifier';
+import truncateMiddle from '@cardstack/safe-tools-client/helpers/truncate-middle';
+import cssVar from '@cardstack/boxel/helpers/css-var';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { later } from '@ember/runloop';
+import copyToClipboard from '@cardstack/boxel/helpers/copy-to-clipboard';
+import didInsert from '@ember/render-modifiers/modifiers/did-insert';
+import { task } from 'ember-concurrency-decorators';
+import perform from 'ember-concurrency/helpers/perform';
+
+import './index.css';
+
+interface Signature {
+  Element: HTMLDivElement;
+  Args: {
+    address: string;
+    isCopyable?: boolean | undefined;
+    copyIconColor?: string;
+  }
+}
+
+export default class TruncatedBlockchainAddress extends Component<Signature> {
+  @tracked isShowingCopiedConfirmation = 0;
+
+  @action registerListener(element: Element) {
+    element.addEventListener('click', (e) => {
+      e.stopPropagation();
+      return false;
+    });
+  }
+
+  @task *flashCopiedConfirmation() {
+    this.isShowingCopiedConfirmation = 1;
+    later(() => {
+      this.isShowingCopiedConfirmation = 0;
+    } , 1000)
+  }
+
+  <template>
+    {{#if @isCopyable}}
+      {{!@glint-expect-error "Passing this function should be valid but it is not as of 1.0.0-beta.1"}}
```

Comment:
Why is it not valid?

Reply:
I am unsure about the error here, it's a weird error. I fixed it the same way as we did [here](https://github.com/cardstack/cardstack/pull/3467#issuecomment-1335834902)